packages/mobile/src/navigator/DrawerNavigator.tsx

```diff
@@ -36,13 +46,79 @@ import { Help } from 'src/icons/navigator/Help'
 import { Home } from 'src/icons/navigator/Home'
 import { Settings } from 'src/icons/navigator/Settings'
 import { useDollarsToLocalAmount, useLocalCurrencySymbol } from 'src/localCurrency/hooks'
+import { ensurePincode } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import useSelector from 'src/redux/useSelector'
 import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
+import Logger from 'src/utils/Logger'
 import { currentAccountSelector } from 'src/web3/selectors'
 
+const TAG = 'NavigationService'
+
 const Drawer = createDrawerNavigator()
 
+type CustomDrawerItemListProps = Omit<DrawerContentOptions, 'contentContainerStyle' | 'style'> & {
+  state: DrawerNavigationState
+  navigation: DrawerNavigationHelpers
+  descriptors: DrawerDescriptorMap
+  protectedRoutes: string[]
+}
+
+function CustomDrawerItemList({
+  state,
+  navigation,
+  descriptors,
+  activeTintColor,
+  inactiveTintColor,
+  activeBackgroundColor,
+  inactiveBackgroundColor,
+  itemStyle,
+  labelStyle,
+  protectedRoutes,
+}: CustomDrawerItemListProps) {
+  const buildLink = useLinkBuilder()
+
+  return (state.routes.map((route, i) => {
+    const focused = i === state.index
+    const { title, drawerLabel, drawerIcon } = descriptors[route.key].options
+    const navigateToItem = () => {
+      navigation.dispatch({
+        ...(focused ? DrawerActions.closeDrawer() : CommonActions.navigate(route.name)),
+        target: state.key,
+      })
+    }
+    const onPress = () => {
+      if (protectedRoutes.indexOf(route.name) !== -1) {
+        // Route should be protected by PIN code
+        ensurePincode()
+          .then(navigateToItem)
+          .catch((error) => {
+            Logger.error(`${TAG}@onPress`, 'PIN ensure error', error)
+          })
+      } else {
+        navigateToItem()
+      }
+    }
+
+    return (
+      <DrawerItem
+        key={route.key}
+        label={drawerLabel !== undefined ? drawerLabel : title !== undefined ? title : route.name}
+        icon={drawerIcon}
+        focused={focused}
+        activeTintColor={activeTintColor}
```

Comment:
Is there a more succinct way to do this passthrough, like maybe `{...props}` instead of deconstruction all these props and then passing them through manually?